tests/unit/src/util/file.ts

```diff
@@ -0,0 +1,77 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import { stub, SinonStub } from 'sinon';
+
+let module: any;
+let existsSyncStub: SinonStub;
+let mkdtempSyncStub: SinonStub;
+let syncStub: SinonStub;
+let joinStub: SinonStub;
+
+registerSuite({
+	name: 'util/file',
+
+	before() {
+		existsSyncStub = stub();
+		mkdtempSyncStub = stub();
+		syncStub = stub();
+		joinStub = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		module = loadModule('src/util/file', {
+			'fs': {
+				existsSync: existsSyncStub,
+				mkdtempSync: mkdtempSyncStub
+			},
+			'mkdirp': {
+				sync: syncStub
+			},
+			'path': {
+				join: joinStub
+			}
+		});
+	},
+
+	afterEach() {
+		existsSyncStub.reset();
+		mkdtempSyncStub.reset();
+		syncStub.reset();
+		joinStub.reset();
+	},
+
+	makeTempDirectory: {
+		'base directory should be created if it does not exist'() {
```

Comment:
For consistency I recommend labeling tests as "unit under test", "condition", "result"

Can we rename these tests to follow this convention?
e.g. "base directory does not exist; directory is created"

I know it can seem a bit odd to write but the consistency reads better and it's easier to compose conditions if multiple suites are used